Multiple regressions introduced in gym 0.21

## 🐛 Bug

Failing [CI job](https://github.com/facebookresearch/CompilerGym/pull/454/checks?check_run_id=3829750198):

```
=================================== FAILURES ===================================
_______________________________ test_basic[cpu] ________________________________
[gw0] darwin -- Python 3.9.7 /Users/runner/hostedtoolcache/Python/3.9.7/x64/bin/python

backend = 'cpu'

    @pytest.mark.parametrize("backend", lt.backends())
    def test_basic(backend):
>       with compiler_gym.make("loop_tool-v0") as env:

tests/loop_tool/actions_test.py:16: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/compiler_gym/util/registration.py:16: in make
    return gym.make(id, **kwargs)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/gym/envs/registration.py:235: in make
make: *** [install-test-cov] Error 1
    return registry.make(id, **kwargs)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/gym/envs/registration.py:129: in make
    env = spec.make(**kwargs)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/gym/envs/registration.py:93: in make
    spec = copy.deepcopy(self)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/copy.py:172: in deepcopy
    y = _reconstruct(x, memo, *rv)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/copy.py:270: in _reconstruct
    state = deepcopy(state, memo)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/copy.py:146: in deepcopy
    y = copier(x, memo)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/copy.py:230: in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/copy.py:146: in deepcopy
    y = copier(x, memo)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/copy.py:230: in _deepcopy_dict
    y[deepcopy(key, memo)] = deepcopy(value, memo)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/copy.py:146: in deepcopy
    y = copier(x, memo)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/copy.py:205: in _deepcopy_list
    append(deepcopy(a, memo))
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/copy.py:172: in deepcopy
    y = _reconstruct(x, memo, *rv)
/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/copy.py:272: in _reconstruct
    y.__setstate__(state)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError('id') raised in repr()] FLOPSReward object at 0x113ac9d40>
state = ({'name': 'flops', 'previous_flops': None}, {'default_negates_returns': True, 'default_value': 0, 'deterministic': False, 'dtype': <class 'numpy.float64'>, ...})

    def __setstate__(self, state):
        # Don't mutate the original state
>       state = dict(state)
E       ValueError: dictionary update sequence element #1 has length 10; 2 is required

/Users/runner/hostedtoolcache/Python/3.9.7/x64/lib/python3.9/site-packages/gym/spaces/space.py:65: ValueError

---------- coverage: platform darwin, python 3.9.7-final-0 -----------
Coverage XML written to file /Users/runner/work/CompilerGym/CompilerGym//coverage.xml

============================= slowest 5 durations ==============================
3.51s call     tests/loop_tool/actions_test.py::test_rand[cpu]
3.10s call     tests/loop_tool/actions_test.py::test_basic[cpu]
3.07s call     tests/loop_tool/actions_test.py::test_induced_remainder[cpu]
2.38s call     tests/loop_tool/actions_test.py::test_thread_removal[cpu]
2.02s call     tests/loop_tool/actions_test.py::test_thread_addition[cpu]
=========================== short test summary info ============================
FAILED tests/loop_tool/actions_test.py::test_basic[cpu] - ValueError: diction...
FAILED tests/loop_tool/actions_test.py::test_induced_remainder[cpu] - ValueEr...
FAILED tests/loop_tool/actions_test.py::test_rand[cpu] - ValueError: dictiona...
FAILED tests/loop_tool/actions_test.py::test_thread_removal[cpu] - ValueError...
FAILED tests/loop_tool/actions_test.py::test_thread_addition[cpu] - ValueErro...
============================== 5 failed in 16.74s ==============================
```

--- a/compiler_gym/requirements.txt
+++ b/compiler_gym/requirements.txt
@@ -3,7 +3,7 @@ deprecated>=1.2.12
 docker>=4.0.0
 fasteners>=0.15
 grpcio>=1.32.0
-gym>=0.18.0
+gym>=0.18.0,<0.21
 humanize>=2.6.0
 loop_tool_py==0.0.7
 networkx>=2.5

